File: javascript/gpt/p73.js
/**
 * p73:
 * - Demonstrates allocating a "person" object and accessing members via reference.
 */

function allocatePerson() {
  return { status: 0, name: "" };
}

function main() {
  const p = allocatePerson();
  p.status = 1;
  p.name = "alice";
  console.log(`name=${p.name} status=${p.status}`);
}

main();

File: python/gpt/p73.py
"""
p73:
- Demonstrates allocating a 'person' object and accessing members via a reference.
"""


class Person:
    def __init__(self) -> None:
        self.status = 0
        self.name = ""


def allocate_person() -> Person:
    return Person()


def main() -> None:
    p = allocate_person()
    p.status = 1
    p.name = "alice"
    print(f"name={p.name} status={p.status}")


if __name__ == "__main__":
    main()
